code/content_packages/valsalia/items/egg_yinglet.dm

```diff
@@ -0,0 +1,87 @@
+//yinglet nests and yinglet eggs
+
+//nest
+/obj/structure/yinglet_nest
+	name = "eggbed"
+	icon = 'code/content_packages/valsalia/icons/egg_yinglet.dmi'
+	icon_state = "base"
+	w_class = ITEM_SIZE_HUGE
+	var/has_egg = TRUE
+
+/obj/structure/yinglet_nest/empty
+	has_egg = FALSE
+
+/obj/structure/yinglet_nest/Initialize()
+	. = ..()
+	if(has_egg)
+		new /obj/item/yinglet_egg(src)
+	update_icon()
+	update_desc()
+
+/obj/structure/yinglet_nest/proc/update_desc()
+	desc = "A pile of plain but very soft fabric arranged to carefully hold an egg, as well as prevent the newly hatched younglet from smashing their face when they flop out."
+	if(has_egg && contents.len)
+		var/obj/item/I = locate() in contents
```

Comment:
Possibly `pick(contents)`? There's a very, very slight edge case where there could be a /mob in the egg bed but not an /obj/item.